src/system.ts

```diff
@@ -239,8 +265,9 @@ export class System extends Element {
     this.partStaves.forEach((part) => part.stave.setNoteStartX(startX));
     if (this.options.autoWidth && this.partVoices.length > 0) {
       justifyWidth = formatter.preCalculateMinTotalWidth(this.partVoices);
+      this.options.width = justifyWidth + Stave.rightPadding + (startX - this.options.x);
       this.partStaves.forEach((part) => {
-        part.stave.setWidth(justifyWidth + Stave.rightPadding + (startX - this.options.x));
+        part.stave.setWidth(this.options.width);
```

Comment:
Kind of a picky point, but I feel like `options` isn't the right place to store this value.  options is stuff set in the constructor.  Probably what makes more sense is to change how bounding box is computed if autowidth is set.

```
let computedWidth = this.options.width;
if (this.options.autoWidth && this.partVoices.length > 0) {
  computedWidth = ...
} else { ...
}

```

And then use `computedWidth` for bounding box.  The logic here all looks correct, though.

Reply:
@AaronDavidNewman Thanks for the review!
I started with a similar approach to your suggestion making the computation close to
```js
this.boundingBox = new BoundingBox(this.options.x, this.options.y, this.options.width, this.lastY - this.options.y);
```
but I decided to change for the following reasons:
- the calculation was duplicating code already there to calculate `justifyWidth`
- `set/getX` `set/getY` had to store the value in `this.options` to make sure that the values are used in `format`
- `setOptions` is a public method. So the values can be changed after construction
- when `options.autoWidth` is set `options.width`is set with a 500 (default value) which is of no use

The logic works! I have a PR prepared for vexml with the following code inspired on your suggestion in https://github.com/0xfe/vexflow/issues/1411

```js
let prevSystem: VF.System | undefined;
systems.forEach((s) => {
  if (prevSystem) {
    let x = prevSystem.getX() + prevSystem.getBoundingBox()!.getW();
    let y = prevSystem.getY();
    if (x > 1000) {
      x = 0;
      y += prevSystem.getBoundingBox()!.getH() + 50;
      s.addConnector('singleLeft');
    } 
    s.setX(x);
    s.setY(y);
  } else {
    s.addConnector('singleLeft');
  }
  s.format();
  prevSystem = s;
});
factory.draw());

```

Reply:
@AaronDavidNewman I believe that the line 217 has a bug but I left it untouched:
https://github.com/0xfe/vexflow/blob/3cfc2f6f8dd6c807177276549f9a087aabb509e8/src/system.ts#L216-L218
I believe that it should be:
```js
justifyWidth = this.options.noPadding
  ? this.options.width - (startX - this.options.x)
  : this.options.width - (startX - this.options.x) - Stave.defaultPadding;
```
what do you think?

Reply:
I also left this alone.  I don't understand the use case for 'no padding', if it is only about the right padding then your way is correct.

Reply:
I changed it the current calculation could result in negative numbers

Reply:
I am not sure, I leave it as it is.